auth-server/src/resolvers/mutation/addressSignupConfirm.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright 2019-2020 @paritytech/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+import { UserInputError } from 'apollo-server';
+import { uuid } from 'uuidv4';
+
+import AuthService from '../../services/auth';
+import { AddressSignupConfirmArgs, Context, SignUpResultType } from '../../types';
+import messages from '../../utils/messages';
+import setRefreshTokenCookie from '../../utils/setRefreshTokenCookie';
+import validateEmail from '../../utils/validateEmail';
+
+export default async (
+	parent: void,
+	{
+		address,
+		email,
+		name,
+		network,
+		signature,
+		username
+	}: AddressSignupConfirmArgs,
+	ctx: Context
+): Promise<SignUpResultType> => {
+	if (email && !validateEmail(email)) {
+		throw new UserInputError(messages.INVALID_EMAIL);
+	}
+
+	username = username || uuid().split('-').join('').substring(0, 25);
```

Comment:
We can move this logic to `AddressSignupConfirm`, before adding the user to the DB.

Reply:
Done